src/core/PortalManager.cpp

```diff
@@ -192,6 +192,11 @@ void CPortalManager::onGlobal(uint32_t name, const char* interface, uint32_t ver
         if (!std::any_cast<Hyprlang::INT>(m_sConfig.config->getConfigValue("general:toplevel_dynamic_bind")))
             m_sHelpers.toplevel->activate();
     }
+
+    else if (INTERFACE == hyprland_toplevel_mapping_manager_v1_interface.name) {
```

Comment:
you don't handle cases in the codebase where this is never called (e.g. hyprland 0.48.1). You need to handle them appropriately, likely send HA> of 0.

Reply:
done, I added a null check everywhere where the struct member is accessed